File: wrench-app/src/main/java/com/izettle/wrench/applicationlist/ApplicationsFragment.java
package com.izettle.wrench.applicationlist;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ViewAnimator;

import com.izettle.wrench.databinding.FragmentApplicationsBinding;
import com.izettle.wrench.di.Injectable;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;


public class ApplicationsFragment extends Fragment implements Injectable {

    @Inject
    ViewModelProvider.Factory viewModelFactory;
    private FragmentApplicationsBinding fragmentApplicationsBinding;

    public ApplicationsFragment() {
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        fragmentApplicationsBinding = FragmentApplicationsBinding.inflate(inflater, container, false);
        fragmentApplicationsBinding.list.setLayoutManager(new LinearLayoutManager(getContext()));
        return fragmentApplicationsBinding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        ApplicationViewModel model = ViewModelProviders.of(this, viewModelFactory).get(ApplicationViewModel.class);

        ApplicationAdapter adapter = new ApplicationAdapter();
        model.getApplications().observe(this, adapter::submitList);
        fragmentApplicationsBinding.list.setAdapter(adapter);

        model.isListEmpty().observe(this, isEmpty -> {
            ViewAnimator animator = fragmentApplicationsBinding.animator;
            if (isEmpty == null || isEmpty) {
                animator.setDisplayedChild(animator.indexOfChild(fragmentApplicationsBinding.noApplicationsEmptyView));
            } else {
                animator.setDisplayedChild(animator.indexOfChild(fragmentApplicationsBinding.list));
            }
        });

    }


}

File: wrench-app/src/main/java/com/izettle/wrench/applicationlist/ApplicationsFragment.kt
package com.izettle.wrench.applicationlist

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.izettle.wrench.databinding.FragmentApplicationsBinding
import org.koin.androidx.viewmodel.ext.viewModel


class ApplicationsFragment : Fragment() {

    private val model: ApplicationViewModel by viewModel()

    private lateinit var fragmentApplicationsBinding: FragmentApplicationsBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        fragmentApplicationsBinding = FragmentApplicationsBinding.inflate(inflater, container, false)
        fragmentApplicationsBinding.list.layoutManager = LinearLayoutManager(context)
        return fragmentApplicationsBinding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        fragmentApplicationsBinding.list.layoutManager = LinearLayoutManager(requireContext())

        val adapter = ApplicationAdapter()
        model.applications.observe(this, Observer { adapter.submitList(it) })
        fragmentApplicationsBinding.list.adapter = adapter

        model.isListEmpty.observe(this, Observer { isEmpty ->
            val animator = fragmentApplicationsBinding.animator
            if (isEmpty == null || isEmpty) {
                animator.displayedChild = animator.indexOfChild(fragmentApplicationsBinding.noApplicationsEmptyView)
            } else {
                animator.displayedChild = animator.indexOfChild(fragmentApplicationsBinding.list)
            }
        })
    }
}
